File: My 120+ Solutions!/Group Anagrams/Group_Anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if (strs.length == 0){
            return new ArrayList();
        }
        HashMap<String, List<String>> hashmap = new HashMap<>();
        for (String s : strs){
            char [] charArray = s.toCharArray();
            Arrays.sort(charArray);
            String key = new String(charArray);
            if (hashmap.containsKey(key)){
                hashmap.get(key).add(s);
            }
            else {
                List<String> l = new ArrayList<>();
                l.add(s);
                hashmap.put(key, l);
            }
        }
        return new ArrayList(hashmap.values());
    }
}

File: Coding Problems & My Solutions/Group Anagrams/Group_Anagrams.swift
class Solution {
    func groupAnagrams(_ strs: [String]) -> [[String]] {
        var ans: [[String]] = []
        var dict: [String:[String]] = [:]
        for str in strs {
            let key: String = String(str.sorted())
            var values: [String] = dict[key] ?? []
            values.append(str)
            dict[key] = values
        }
        for pair in dict {
            let group: [String] = pair.value
            ans.append(group)
        }
        return ans
    }
}